test/oneof.test.js

```diff
@@ -0,0 +1,399 @@
+'use strict'
+
+const { test } = require('tap')
+const build = require('..')
+
+test('object with multiple types field', (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'object with multiple types field',
+    type: 'object',
+    properties: {
+      str: {
+        oneOf: [{
+          type: 'string'
+        }, {
+          type: 'boolean'
+        }]
+      }
+    }
+  }
+  const stringify = build(schema)
+
+  try {
+    const value = stringify({
+      str: 'string'
+    })
+    t.is(value, '{"str":"string"}')
+  } catch (e) {
+    console.log('errorrrrr', e)
+    t.fail()
+  }
+
+  try {
+    const value = stringify({
+      str: true
+    })
+    t.is(value, '{"str":true}')
+  } catch (e) {
+    t.fail()
+  }
+})
+
+test('object with field of type object or null', (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'object with field of type object or null',
+    type: 'object',
+    properties: {
+      prop: {
+        oneOf: [{
+          type: 'object',
+          properties: {
+            str: {
+              type: 'string'
+            }
+          }
+        }, {
+          type: 'null'
+        }]
+      }
+    }
+  }
+  const stringify = build(schema)
+
+  try {
+    const value = stringify({
+      prop: null
+    })
+    t.is(value, '{"prop":null}')
+  } catch (e) {
+    t.fail()
+  }
+
+  try {
+    const value = stringify({
+      prop: {
+        str: 'string'
```

Comment:
This will check that we are removing also the additional fields
```suggestion
        str: 'string',
        remove: 'this'
```